examples/auction/auction_cc.cpp

```diff
@@ -48,8 +48,18 @@ int invoke(const char* args,
     }
     else if (function_name == "submit")
     {
-        std::string bidder_name = argss[2];
         int value = std::stoi(argss[3]);
+        std::string bidder_name = argss[2];
+        // TODO: eventually replace bidder_name with get_creator_name but for now
+        //  in our tests we have only one client, so leave passed bidder_name to
+        //  allow for different bidders ...
+        char real_bidder_name_msp_id[1024];
+        char real_bidder_name_dn[1024];
+        get_creator_name(real_bidder_name_msp_id, sizeof(real_bidder_name_msp_id),
```

Comment:
This input is not validated.
Most importantly, I don't see an easy way to do that. Even if you can verify the MSP id and the certificate, it does not mean that you have the creator's name.

Reply:
Yes, the current implementation is not secure (as is many other things right now :-) and the goal is primarily to fix API.  I should have added a corresponding TODO in shim.ccp, though, which i just did.
Regarding implementation: There is definition no easy-way (or any?) with the current way of piggy-backing on top of the go-shim but with a proper chaincode type we can have access to the signed proposal in the enclave (see above-referenced TODO). The verification of the whole thing is certainly non-trivial but we more or less have to do that anyway inside TLCC so we can just off-load it to TLCC.  Lastly, the creator name is by definition the distinguished name in the certificate, so if we have the certificate, we have the name ...